uber/models/department.py

```diff
@@ -0,0 +1,271 @@
+    'job_required_job_role', 'Department', 'DepartmentMembership', 'Job',
+    'JobRole', 'Shift']
+
+
+department_membership_job_role = Table(
+    'department_membership_job_role',
+    MagModel.metadata,
+    Column('department_membership_id', UUID,
+           ForeignKey('department_membership.id')),
+    Column('job_role_id', UUID, ForeignKey('job_role.id')))
+
+
+department_membership_request = Table(
+    'department_membership_request',
+    MagModel.metadata,
+    Column('attendee_id', UUID, ForeignKey('attendee.id')),
+    Column('department_id', UUID, ForeignKey('department.id'), nullable=True),
+    UniqueConstraint('attendee_id', 'department_id'))
+
+
+job_required_job_role = Table(
+    'job_required_job_role',
+    MagModel.metadata,
+    Column('job_id', UUID, ForeignKey('job.id')),
+    Column('job_role_id', UUID, ForeignKey('job_role.id')))
+
+
+class DepartmentMembership(MagModel):
+
+    DEPARTMENT_HEAD = 100
+    TRUSTED = 200
+    REGULAR = 300
+    BANNED = 500
+
+    STATUS_OPTS = [
+        (DEPARTMENT_HEAD, 'Department Head'),
+        (TRUSTED, 'Trusted'),
+        (REGULAR, 'Regular'),
+        (BANNED, 'Banned')]
+    STATUSES = dict(STATUS_OPTS)
+
+    status = Column(Choice(STATUS_OPTS), default=REGULAR)
+    attendee_id = Column(UUID, ForeignKey('attendee.id'))
+    department_id = Column(UUID, ForeignKey('department.id'))
+
+    job_roles = relationship(
+        'JobRole',
+        backref='department_memberships',
+        secondary='department_membership_job_role',
+        cascade=False)
+
+    __table_args__ = (UniqueConstraint('attendee_id', 'department_id'),)
+
+
+class Department(MagModel):
+    name = Column(UnicodeText)
+    description = Column(UnicodeText)
+    accepts_volunteers = Column(Boolean, default=True)
+    parent_id = Column(UUID, ForeignKey('department.id'), nullable=True)
+
+    jobs = relationship('Job', backref='department')
+    job_roles = relationship('JobRole', backref='department')
+    department_heads = relationship(
+        'Attendee',
+        backref='headed_depts',
+        primaryjoin='and_('
+                    'Department.id==DepartmentMembership.department_id, '
+                    'DepartmentMembership.status=={})'.format(
+                        DepartmentMembership.DEPARTMENT_HEAD),
+        secondaryjoin='DepartmentMembership.attendee_id==Attendee.id',
+        secondary='department_membership',
+        viewonly=True,
+        cascade=False)
+    members = relationship(
+        'Attendee',
+        backref='assigned_depts',
+        secondary='department_membership',
+        cascade=False)
+    membership_requests = relationship(
+        'Attendee',
+        backref='requested_depts',
+        secondary='department_membership_request',
+        cascade=False)
+    sub_departments = relationship(
+        'Department',
+        backref='parent',
+        remote_side='Department.id',
+        single_parent=True)
+    trusted_members = relationship(
```

Comment:
Same point as before -- "trusted" was, from a usage point of view, just a very basic way of sorting jobs into two different roles.

Reply:
Agree!